api/src/Serializer/Denormalizer/InputFilterDenormalizer.php

```diff
@@ -63,10 +64,27 @@ protected function filterInputs($data, string $className): array {
 
         $reflClass = $this->getReflectionClass($className);
         $filterAttributes = [];
+        $toFilterRelatedEntities = [];
         foreach ($reflClass->getProperties() as $property) {
+            $propertyName = $property->name;
             if ($property->getDeclaringClass()->name === $className) {
                 foreach ($this->getInputFilterAttributes($property) as $filterAttribute) {
-                    $filterAttributes[] = [$property->name, $filterAttribute];
+                    $filterAttributes[] = [$propertyName, $filterAttribute];
+                }
+            }
+            $reflectionUnionType = $property->getType();
+            if ($reflectionUnionType instanceof \ReflectionNamedType) {
+                $propertyClassName = $reflectionUnionType->getName();
+                if (is_a($propertyClassName, BaseEntity::class, true)) {
+                    $toFilterRelatedEntities[] = [$propertyName, $propertyClassName];
+                }
+            } elseif ($reflectionUnionType instanceof \ReflectionUnionType) {
+                foreach ($reflectionUnionType->getTypes() as $type) {
+                    if (is_a($type->getName(), BaseEntity::class, true)) {
+                        $toFilterRelatedEntities[] = [$propertyName, $type];
+
+                        break;
+                    }
```

Comment:
Was macht der Code oben genau? Docu/comments möglich?

Reply:
na ja.
Man könnte es in eine Methode auslagern, und dann
```
$filterAttributes[] = [$property->name, $filterAttribute];
foreach ($reflClass->getProperties() as $property) {
            if ($property->getDeclaringClass()->name === $className) {
                foreach ($this->getInputFilterAttributes($property) as $filterAttribute) {
                    $filterAttributes[] = [$propertyName, $filterAttribute];
                }
            }
            $tuple = getTupleForRelationProperty($property)
            if($tuple != null) {
                $toFilterRelatedEntities[] = $tuple;
             }
```

Oder sonst müssen die tests reichen..?

Reply:
Aus security sicht möchte ich eigentlich InputFilter und Validations nicht einschalten müssen.
Sondern falls jemand einen guten grund hat, das auszuschalten, kann er das. Aber per default werden Validations und Inputfilter aktiviert und überprüft.

Ich würde das eher noch ausbauen, dass es das automatisch für collections auch noch macht (mit der template Annotation/Attribute hoffentlich). Und dann das in 3 verschiedene Denormalizer packen (1 Denormalizer für eine Entity, 1 Denormalizer für OneToOne und ManyToOne, 1 denormalizer für ManyToOne und ManyToMany, und evtl später noch für Embeddable Entities, falls wir das mal brauchen.)

Das aufteilen auf 2 loops mache ich gerne noch.

Reply:
> Soweit ich sehe ist der neue Code dafür zuständig, auch die InputFilters in related Entities zu evaluieren.

Ah, es geht um denormalizing von embedded entities (writableLink: true). Das war mir jetzt beim Überfliegen des Codes nicht klar. Ein paar Kommentarzeilen wären hilfreich.

Ansonsten mit Kommentar von @carlobeltrame good for me.

Reply:
Ist ausgelagert in eine separate methode